test/test_torch.py

```diff
@@ -5483,6 +5483,98 @@ def test_dot_data_use(self):
             model.weight.data = weight
             out = model(input)
 
+    def test_bucketization(self):
+        for device in torch.testing.get_all_device_types():
+            values_1d = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8, 9], device=device)
+            values_3d = torch.tensor([[[1, 3, 5], [2, 4, 6]], [[1, 2, 3], [4, 5, 6]]], device=device)
+
+            # regular case 3d boundary and 3d input value
+            boundaries = torch.tensor([[[1, 2, 3, 4], [3, 4, 5, 6]], [[1, 3, 5, 7], [2, 4, 6, 8]]], device=device)
+            result = torch.tensor([[[0, 2, 4], [0, 1, 3]], [[0, 1, 1], [1, 2, 2]]], device=device)
+            output = torch.zeros(2, 2, 3, device=device, dtype=torch.int64)
+            self.assertEqual(torch.equal(torch.searchsorted(boundaries, values_3d), result), True)
```

Comment:
You can directly compare tensors with self.assertEqual

Reply:
done